src/updater/moduleUpdater.js

```diff
@@ -131,27 +122,22 @@ const downloadModule = async (name, ver) => {
   downloading.total++;
 
   const path = join(downloadPath, name + '-' + ver + '.zip');
-  const file = fs.createWriteStream(path);
 
   // log('Modules', 'Downloading', `${name}@${ver}`);
 
-  let success, total, cur =  0;
-  const res = await redirs(baseUrl + '/' + name + '/' + ver + qs);
-  success = res.statusCode === 200;
-  total = parseInt(res.headers['content-length'] ?? 1, 10);
-
-  res.pipe(file);
+  let success, total, cur = 0;
+  const res = await get(baseUrl + '/' + name + '/' + ver + qs);
+  success = res[0] === 200;
 
-  res.on('data', c => {
-    cur += c.length;
-
-    events.emit('downloading-module', { name, cur, total });
-  });
-
-  await new Promise((res) => file.on('close', res));
-
-  if (success) commitManifest();
-    else downloading.fail++;
+  // todo: if a progress-bar-like interface and stream-like file writing are still wanted, implement
+  if (success) {
+    total = parseInt(res[2].get('content-length') ?? 1, 10);
+    events.emit('downloading-module', { name, total, total });
+    fs.writeFile(path, res[1], e => log('ModuleUpdate', 'Writing to file failed:', e));
```

Comment:
the callback still gets called if there is no error so this will always log a fail?